js/runner.go

```diff
@@ -343,13 +350,14 @@ func (u *VU) RunOnce(ctx context.Context) error {
 		}()
 	}
 
-	// Lazily JS-ify setupData on first run. This is lightweight enough that we can get away with
-	// it, and alleviates a problem where setupData wouldn't get populated properly if NewVU() was
-	// called before Setup(), which is hard to avoid with how the Executor works w/o complicating
-	// the local executor further by deferring SetVUsMax() calls to within the Run() function.
-	if u.setupData == nil && u.Runner.setupData != nil {
-		u.setupData = u.Runtime.ToValue(u.Runner.setupData)
+	// Always unmarshall the setupData so that it doesn't change between calls
```

Comment:
Yep, shouldn't be an issue, since there couldn't be any data races that way. I think we shouldn't advertise it in the docs, since we might want to change it in the future, but it should be fine - other things in VUs stick around between iterations as well.